### PR TITLE
fix: Org profile slug [CAL-2252]

### LINKED ISSUE #10371: [CAL-2252] Team name is empty
It very often doesn't load any of the data here

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2252](https://linear.app/calcom/issue/CAL-2252/team-name-is-empty)</sub>

### PR DESCRIPTION
**Pull request by leog**
## What does this PR do?

Org slug could be undefined if it's an unpublished org, relying on `metadata.requestedSlug` as a fallback.

Fixes #10371 

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

Having orgs enabled and creating one without paying for subscription will produce an unpublished org. Going to the profile page fo that unpublished org should show the correct name and slug for the org.

### CODE REVIEW

File: packages/features/ee/organizations/pages/settings/profile.tsx [hunk 1]
```diff
@@ -89,20 +88,8 @@ const OrgProfileView = () => {
...
     !currentOrganisation.bio.replace("<p><br></p>", "").length;
 
-  const deleteTeamMutation = trpc.viewer.teams.delete.useMutation({
-    async onSuccess() {
-      await utils.viewer.teams.list.invalidate();
-      showToast(t("your_org_disbanded_successfully"), "success");
-      router.push(`${WEBAPP_URL}/teams`);
-    },
-  });
-
```
Comment (leog) [inline packages/features/ee/organizations/pages/settings/profile.tsx:99]: Unused code

File: packages/features/ee/organizations/pages/settings/profile.tsx [hunk 2]
```diff
@@ -89,20 +88,8 @@ const OrgProfileView = () => {
...
   if (!orgBranding) return null;
 
-  function deleteTeam() {
-    if (currentOrganisation?.id) deleteTeamMutation.mutate({ teamId: currentOrganisation.id });
-  }
-
```
Comment (leog) [inline packages/features/ee/organizations/pages/settings/profile.tsx:105]: Unused code

File: packages/features/ee/organizations/pages/settings/profile.tsx [hunk 3]
```diff
@@ -168,15 +155,21 @@ const OrgProfileView = () => {
...
+                render={({ field: { value } }) => (
+                  <div className="mt-8">
+                    <TextField
+                      name="slug"
+                      label={t("org_url")}
+                      value={value}
+                      disabled
+                      addOnSuffix={`.${subdomainSuffix()}`}
+                    />
+                  </div>
```
Comment (leog) [inline packages/features/ee/organizations/pages/settings/profile.tsx:170]: Grabbing form value from setted value after trpc returns org

File: packages/trpc/server/routers/viewer/organizations/list.handler.ts [hunk 4]
```diff
@@ -28,11 +29,14 @@ export const listHandler = async ({ ctx }: ListHandlerInput) => {
...
 
+  const metadata = teamMetadataSchema.parse(membership?.team.metadata);
+
   return {
     user: {
       role: membership?.role,
       accepted: membership?.accepted,
     },
     ...membership?.team,
+    metadata,
```
Comment (leog) [inline packages/trpc/server/routers/viewer/organizations/list.handler.ts:40]: Providing the parsed metadata to access `requestedSlug`

**Review by Udit-takkar — COMMENTED** [review]
Your changes look fine but i am not able to create a new team after creating a org.

I get 404 at this step

Screenshot 2023-08-03 at 2 31 51 PM

and this team's profile page just keeps loading

**Comment by leog** [comment]
> Your changes look fine but i am not able to create a new team after creating a org.
> and this team's profile page just keeps loading

Thanks @Udit-takkar. Please try again, the fix for that was pushed in a different PR that is merged now.

**Review by Udit-takkar — COMMENTED** [review]
@leog this works fine but as a user i am confused if i do not publish a team then it still shown in /settings as a normal team and how do i publish it?

**Review by Udit-takkar — APPROVED** [review]
LGTM

**Review by zomars — APPROVED** [review]
LGTM

**Pull request merged**